### ISSUE TITLE
Bug in User.php::__toString() function

### ISSUE THREAD

**Issue by Mozan** [contao/core-bundle#513]
Sorry - no time for a PR.

```php
function __toString()
```
assume that there is always a valid

```php
$this->username 
```
Which is not always true.
Simple fix:
```php
return $this->intId ? (($this->username) ? $this->username : 'anon.') : 'anon.';
```

**Comment by aschempp** [comment]
So you think we have time for your PR because you do not? 👎 

The mentioned code is already there: https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/User.php#L218

**Comment by leofeyer** [comment]
> Also, *anon.* would certainly not be a correct representation of a logged in user.

I second this.

**Comment by Mozan** [comment]
I never implied anyone of you need to fix it ASAP. 
Just wanted to highlight the problem.

Just look at: https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/User.php#L218

There is `'anon'` in the code- i didn't invent it :)
And - as you know - we can easly remove `username` from tl_member (as I did in my project) - and - boom - there's the error. 

Original code:
```php
public function __toString()
{
	return $this->intId ? $this->username : 'anon.';
}
```

My proposed simple fix - not sure if proper:
```php
public function __toString()
{
	return $this->intId ? (($this->username) ? $this->username : 'anon.') : 'anon.';
}
```

**Comment by aschempp** [comment]
here's my suggestion then:

```php
public function __toString()
{
    if (!$this->intId) {
        return 'anon.';
    }

    return $this->username ?: ($this->getTable() . '.' . $this->intId);
}
´´´

**Comment by Mozan** [comment]
Looks good to me.
Next time I'll try be much more verbose :)

**Comment by leofeyer** [comment]
Fixed in cc82046ffff2bd93bd4ac69648cbbe354cc2f77f then.
